src/QueryDescription/index.js

```diff
@@ -340,13 +365,51 @@ export function buildQueryDescription(clauses: Clause[]): QueryDescription {
   return query
 }
 
+const whereNotDeleted = where(syncStatusColumn, notEq('deleted'))
+const whereNotDeletedJoin: (TableName<any>) => On = table =>
+  on(table, syncStatusColumn, notEq('deleted'))
+
+function conditionsAndJoinTablesWithoutDeleted(wheres: Where[]): Where[] {
+  const conditions: Where[] = wheres.map(queryWithoutDeletedImpl)
+  // $FlowFixMe
+  const ons: On[] = conditions.filter(clause => clause.type === 'on')
+  const onTables = uniq(ons.map(clause => clause.table))
+  return conditions.concat(onTables.map(whereNotDeletedJoin))
+}
+
+function conditionWithoutDeleted(clause: Where): Where {
+  if (clause.type === 'on') {
+    const onClause: On = clause
+    return {
+      type: 'and',
+      conditions: [clause, whereNotDeletedJoin(onClause.table)],
+    }
+  }
+  return queryWithoutDeletedImpl(clause)
+}
+
+function queryWithoutDeletedImpl(clause: Where): Where {
+  if (clause.type === 'and') {
+    return {
+      type: 'and',
+      conditions: conditionsAndJoinTablesWithoutDeleted(clause.conditions),
```

Comment:
right now, if you have multiple `on` conditions, you'll write it like this:

```
on('projects', foo=bar),
on('projects', bar=baz),
on('projects', baz=blah),
```

so if we treated `and` the same way as `or`, we'd get:

```
and(on('projects', foo=bar), on('projects', _status != 'deleted'))
and(on('projects', bar=baz), on('projects', _status != 'deleted'))
and(on('projects', baz=blah), on('projects', _status != 'deleted'))
```

and since we know we're in an `and`, we can skip the duplicate conditions. They probably wouldn't be harmful for SQLite, but probably would be harmful for performance for LokiJS.

I think the real solution is to introduce this syntax:

```
on('projects', [
  foo = bar,
  bar = baz,
  baz = blah,
])
```

then we wouldn't need this and a lot of other special cases, in addition to making yet other cases more performant on LokiJS :) So this is what I will do - and remove the special cases

Reply:
Thanks for explanation :) So this is optimisation, not that `and` needs to be treated differently? But isn't `or` similar? For:
```
Q.or(
  Q.on("projects", foo=bar),
  Q.on("projects", bar=baz),
  Q.on("projects, baz=blah),
)
```
we get:
```
Q.or(
  Q.and(Q.on("projects", foo=bar), Q.on("projects", _status != "deleted")),
  Q.and(Q.on("projects", bar=baz), Q.on("projects", _status != "deleted")),
  Q.and(Q.on("projects", baz=blah), Q.on("projects", _status != "deleted")),
)
```
right? So ideally we would want sth like this:
```
Q.and(
  Q.or(
    Q.on("projects", foo=bar),
    Q.on("projects", foo=bar),
    Q.on("projects", foo=bar)
  ),
  Q.on("projects", _status != "deleted")
)
```
Or this won't work? New syntax looks nice :)

Reply:
this would work only if all elements of `Q.or` are `Q.on`s on the same table. So we get even more edge cases to cover. Bah, better to fix the root cause